column_index.go

```diff
@@ -354,13 +357,35 @@ func (i *int32ColumnIndexer) IndexPage(numValues, numNulls int64, min, max Value
 	i.maxValues = append(i.maxValues, max.int32())
 }
 
+func reverseInt32MinMaxValues(mLen int, mVal []int32) []byte {
+	buf := make([]byte, mLen*4)
+	idx := 0
+	for k := range mLen {
+		binary.LittleEndian.PutUint32(buf[idx:(4+idx)], uint32(mVal[k]))
+		idx += 4
+	}
+	return buf
+}
+
 func (i *int32ColumnIndexer) ColumnIndex() format.ColumnIndex {
-	return i.columnIndex(
-		splitFixedLenByteArrays(unsafecast.Int32ToBytes(i.minValues), 4),
-		splitFixedLenByteArrays(unsafecast.Int32ToBytes(i.maxValues), 4),
-		orderOfInt32(i.minValues),
-		orderOfInt32(i.maxValues),
-	)
+	if cpu.IsBigEndian {
+		byteMin := reverseInt32MinMaxValues(len(i.minValues), i.minValues)
+		byteMax := reverseInt32MinMaxValues(len(i.maxValues), i.maxValues)
+
+		return i.columnIndex(
+			splitFixedLenByteArrays(byteMin, 4),
+			splitFixedLenByteArrays(byteMax, 4),
+			orderOfInt32(i.minValues),
+			orderOfInt32(i.maxValues),
+		)
+	} else {
+		return i.columnIndex(
+			splitFixedLenByteArrays(unsafecast.Int32ToBytes(i.minValues), 4),
+			splitFixedLenByteArrays(unsafecast.Int32ToBytes(i.maxValues), 4),
+			orderOfInt32(i.minValues),
+			orderOfInt32(i.maxValues),
+		)
+	}
```

Comment:
I would really prefer if we were able to move the implementation of this method to two different files that get selected based on whether we're building for a big-endian or little-endian platform.

We could introduce a convention to have _be.go/_le.go files for endian-dependent code, for example:
```
column_index_be.go
```
```
column_index_le.go
```
Then setting the build tags properly we can make sure that the right version of the code gets selected for each architecture.

Reply:
@achille-roussel
I have made the necessary code changes to take care of the review comment. Code is looking much simplified now. 
Please take a look. Thanks.. !!